src/adapter/src/coord/sequencer.rs

```diff
@@ -1320,18 +1300,22 @@ impl<S: Append + 'static> Coordinator<S> {
         let from_name = from.name().item.clone();
         let from_type = from.item().typ().to_string();
         let result = self
-            .catalog_transact(Some(&session), ops, move |txn| async move {
+            .catalog_transact_with(Some(&session), ops, move |txn| {
                 // Validate that the from collection is in fact a persist collection we can export.
-                txn.dataflow_client
-                    .storage
-                    .collection(sink.from)
-                    .map_err(|e| match e {
-                        StorageError::IdentifierMissing(_) => AdapterError::Unstructured(anyhow!(
+                std::future::ready(
```

Comment:
@jkosh44 this is the trick ^. The closure no longer returns an `async { .. }` block. It does the computation immediately and returns a `Ready` future that contains the result directly